### ISSUE TITLE
d_out

### ISSUE THREAD

**Issue by linwei0763** [leofansq/SCF-Net#14]
Hi Siqi,

Thank you for release your work.

I am tring to use your LFA in my network but am confused about some code.

The dimension of f_lc, shortcut, and f_gc is d_out*2, so makes the returned dimension be d_out*4 in this case. But in RandLA-Net, the output dimension should be d_out*2. Are they different in the architecture? Or should the dimensions of each layer be the same in your network and RandLA-Net?

Appreciate if you could help me with this problem.

in SCFNet.py:
line 276-285

        # MLP 2
        f_lc = helper_tf_util.conv2d(f_lc, d_out * 2, [1, 1], name + 'mlp2', [1, 1], 'VALID', True, is_training, activation_fn=None)
        # MLP Shotcut
        shortcut = helper_tf_util.conv2d(feature, d_out * 2, [1, 1], name + 'shortcut', [1, 1], 'VALID', activation_fn=None, bn=True, is_training=is_training)
        
        # Global Contextual Features
        f_gc = tf.expand_dims(tf.concat([xyz, lg_volume_ratio], axis=-1), -2)
        f_gc = helper_tf_util.conv2d(f_gc, d_out * 2, [1, 1], name + 'lg', [1, 1], 'VALID', activation_fn=None, bn=True, is_training=is_training)

        return tf.nn.leaky_relu(tf.concat([f_lc + shortcut, f_gc], axis=-1))

**Comment by leofansq** [comment]
These are hyper parameters in the network, so you can manually adjust them to fit your own network accordingly. The selection of them need some ablation studies.

**Comment by linwei0763** [comment]
> These are hyper parameters in the network, so you can manually adjust them to fit your own network accordingly. The selection of them need some ablation studies.

Thank you for prompt reply.

I totally agree with you about the selection of these hyperparameters. I am trying to understand your architecture so I really wanna understand each part of your module.

I am using pytorch so I need to specify the input channel number of each MLP or other modules. To my best knowledge, the outout channel number of SCF module is 4*d_out, twice as that of RandLA-Net. So if I want to compare your LFA module and LFA in RandLA-Net by ablation experiments, should I double the output channel number in RandLA-Net? Does it make sense?

Again, Thank you for your help.

BW,
Wei

**Comment by leofansq** [comment]
The dimensions of the Local Contextual feature are the same in the SCFNet and RandLA-Net, while the output dimension of the SCF module is doubled due to the additional Global Contextual feature ([f_lc + shortcut, f_gc]).

**Comment by linwei0763** [comment]
Thank you for your reply. I have tested your modules on my dataset with the same channel numbers (which are half of yours) as RandLA-Net. The results show shat SCF module still outperforms the LFA module in RandLA-Net. Thank you for your work and releasing your codes. 👍
